celements-user-manager/component/src/main/java/com/celements/auth/user/validation/XWikiUsersValidationRule.java

```diff
@@ -0,0 +1,117 @@
+package com.celements.auth.user.validation;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.inject.Inject;
+import javax.validation.constraints.NotNull;
+
+import org.springframework.stereotype.Component;
+import org.xwiki.model.reference.ClassReference;
+
+import com.celements.auth.user.User;
+import com.celements.auth.user.UserService;
+import com.celements.docform.DocFormRequestKey;
+import com.celements.docform.DocFormRequestParam;
+import com.celements.mailsender.IMailSenderRole;
+import com.celements.model.reference.RefBuilder;
+import com.celements.rights.access.EAccessLevel;
+import com.celements.rights.access.IRightsAccessFacadeRole;
+import com.celements.validation.IRequestValidationRule;
+import com.celements.validation.ValidationResult;
+import com.celements.validation.ValidationType;
+
+@Component
+public class XWikiUsersValidationRule implements IRequestValidationRule {
+
+  private IMailSenderRole mailSenderService;
+  private UserService userService;
+  private IRightsAccessFacadeRole rightsAccess;
+
+  @Inject
+  public XWikiUsersValidationRule(IMailSenderRole mailSenderService,
+      UserService userService,
+      IRightsAccessFacadeRole rightsAccess) {
+    this.mailSenderService = mailSenderService;
+    this.userService = userService;
+    this.rightsAccess = rightsAccess;
+  }
+
+  @Override
+  public @NotNull List<ValidationResult> validate(@NotNull List<DocFormRequestParam> params) {
+    Stream<ValidationResult> validationResults = Stream.empty();
+    validationResults = Stream.concat(validationResults,
+        getEmailParam(params).map(this::validate).orElse(Stream.empty()));
+
+    return validationResults.collect(Collectors.toList());
```

Comment:
the local `validationResults` var is not needed:
```suggestion
    return getEmailParam(params)
        .map(this::validate)
        .orElse(Stream.empty()))
        .collect(Collectors.toList());
```